### PR TITLE
Create a full integration test suite

### PR DESCRIPTION
**Pull request by adejanovski**
I've added a full generic test suite in Cucumber to run on real ccm clusters.
Travis build is modified to run those tests on different versions of Cassandra (2.1.16, 2.2.8, 3.0.9 and 3.9 for now).

Made so that the SonarQube analysis only runs in one build and not all four.

### CODE REVIEW

File: src/test/java/com/spotify/reaper/acceptance/BasicSteps.java [hunk 1]
```diff
@@ -299,4 +456,57 @@ public void a_new_daily_repair_schedule_is_added_that_already_exists_for(String
...
+ }
+ 
+ @And("^we wait for at least (\\d+) segments to be repaired$")
+ public void we_wait_for_at_least_segments_to_be_repaired(int nbSegmentsToBeRepaired)
+     throws Throwable {
+   int repairedSegments = 0;
+   int i = 0;
+   
+   while(repairedSegments<nbSegmentsToBeRepaired && i<60){
+     Thread.sleep(10000);
```
Comment (adejanovski) [inline src/test/java/com/spotify/reaper/acceptance/BasicSteps.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)

File: src/test/java/com/spotify/reaper/acceptance/BasicSteps.java [hunk 2]
```diff
@@ -299,4 +456,57 @@ public void a_new_daily_repair_schedule_is_added_that_already_exists_for(String
...
+   String responseData = response.getEntity(String.class);
+   RepairRunStatus run = SimpleReaperClient.parseRepairRunStatusJSON(responseData);
+   TestContext.LAST_MODIFIED_ID = run.getId();
+ }
+ 
+ @And("^we wait for (\\d+) seconds$")
+ public void we_wait_for(int sleepTime)
+     throws Throwable {
+   LOG.info("Sleeping for " + sleepTime + "s...");
+   Thread.sleep(sleepTime * 1000);
```
Comment (adejanovski) [inline src/test/java/com/spotify/reaper/acceptance/BasicSteps.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)

File: pom.xml [hunk 3]
```diff
@@ -116,6 +116,41 @@
                         </configuration>
                     </plugin>
                     <plugin>
+				        <artifactId>maven-surefire-plugin</artifactId>
```
Comment (adejanovski) [inline pom.xml]: MINOR Tab characters found (this is the first occurrence) [rule](https://sonarqube.com/coding_rules#rule_key=xml%3AIllegalTabCheck)

**Comment by adejanovski** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Review by rustyrazorblade — APPROVED** [review]

**Pull request merged**